spec/workflow/runner/docker_spec.rb

```diff
@@ -1,39 +1,101 @@
+
+      subject.run_async!("docker://hello-world:latest", {"FOO" => "BAR"}, {"luggage_password" => "12345"})
+    end
+  end
+
+  describe "#running?" do
+    it "returns true when running" do
+      stub_good_run!("docker", :params => ["inspect", container_id], :output => "[{\"State\": {\"Running\": true}}]")
+      expect(subject.running?(container_id)).to be_truthy
+    end
+
+    it "returns false when completed" do
+      stub_good_run!("docker", :params => ["inspect", container_id], :output => "[{\"State\": {\"Running\": false, \"ExitCode\": 0}}]")
+      expect(subject.running?(container_id)).to be_falsey
+    end
+  end
+
+  describe "#success?" do
+    it "returns true when successful" do
+      stub_good_run!("docker", :params => ["inspect", container_id], :output => "[{\"State\": {\"Running\": false, \"ExitCode\": 0}}]")
+      expect(subject.success?(container_id)).to be_truthy
+    end
+
+    it "returns false when unsuccessful" do
+      stub_good_run!("docker", :params => ["inspect", container_id], :output => "[{\"State\": {\"Running\": false, \"ExitCode\": 1}}]")
+      expect(subject.success?(container_id)).to be_falsey
+    end
+  end
+
+  describe "#output" do
+    it "returns log output" do
+      stub_good_run!("docker", :params => ["logs", container_id], :output => "hello, world!")
+      expect(subject.output(container_id)).to eq("hello, world!")
+    end
+
+    it "raises an exception when getting pod logs fails" do
+      stub_bad_run!("docker", :params => ["logs", container_id])
+      expect { subject.output(container_id) }.to raise_error(AwesomeSpawn::CommandResultError, /docker exit code: 1/)
+    end
+  end
+
+  describe "#cleanup" do
+    let(:secrets_file) { double("Tempfile") }
+
+    it "deletes the container and secret" do
+      stub_good_run!("docker", :params => ["rm", container_id])
+      expect(secrets_file).to receive(:close!)
+      subject.cleanup(container_id, secrets_file)
+    end
+
+    it "doesn't delete the secret_file if not passed" do
+      stub_good_run!("docker", :params => ["rm", container_id])
+      subject.cleanup(container_id, nil)
+    end
 
-      subject.run!("docker://hello-world:latest", {"FOO" => "BAR"}, {"luggage_password" => "12345"})
+    it "deletes the secrets file if deleting the container fails" do
+      stub_bad_run!("docker", :params => ["rm", container_id])
+      expect(secrets_file).to receive(:close!)
+      subject.cleanup(container_id, secrets_file)
     end
 
     context "with network=host" do
       let(:runner_options) { {"network" => "host"} }
 
       it "calls docker run with --net host" do
-        stub_good_run!("docker", :params => ["run", :rm, [:net, "host"], "hello-world:latest"])
+        stub_good_run!("docker", :params => ["run", :rm, "hello-world:latest"])
```

Comment:
This change is confusing - the spec is specifically for when network -> host is being passed, so I would expect it on the command line.

Reply:
This looks like the result of a bad conflict resolution on my part, especially since it is testing `run!` in the `cleanup` describe block.

I've moved this back up to `describe run!` and fixed the missing `[:net, "host"]` (this was the result of redefining subject to not pass runner_options in the spec, not the actual code failing to pass `--net host`)